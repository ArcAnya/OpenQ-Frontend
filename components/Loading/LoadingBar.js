import React from 'react';
import Alert from '../../components/svg/alert';
import Cross from '../../components/svg/cross';

const LoadingBar = ({loadingBar, changeText}) => {
	return (
		<div className='flex fixed bottom-0 z-40 left-0 pl-8 p-4 bg-[#21262d] border border-gray-700 rounded-sm'>
			<div className='flex gap-2 items-center'>
				<Alert />
				<div>{ !changeText?
					'It will take a couple of minutes until your contract will be visible in our explorer'
					: 'Please reload the page to see your new contract in our explorer'
			}</div>

			</div>
			<button className='pl-4 pr-0' onClick={() => loadingBar(false)}>
				<Cross />
			</button>
		</div>
	);
};

export default LoadingBar;