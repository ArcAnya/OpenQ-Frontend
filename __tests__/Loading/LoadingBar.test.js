import React from 'react';
import LoadingBar from '../../components/Loading/LoadingBar';
import renderer from 'react-test-renderer';

describe('LoadingBar', () => {
  it('should match DOM Snapshot', () => {
    const tree = renderer.create(<LoadingBar changeText={false} />);
    expect(tree.toJSON()).toMatchSnapshot('and say "It will take..." ');
  });
  it('should match DOM Snapshot', () => {
    const tree = renderer.create(<LoadingBar changeText={true} />);
    expect(tree.toJSON()).toMatchSnapshot('and say "Please reload..." ');
  });
});
