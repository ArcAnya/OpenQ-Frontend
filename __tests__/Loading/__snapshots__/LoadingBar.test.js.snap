// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingBar should match DOM Snapshot: and say "It will take..."  1`] = `
<div
  className="flex fixed bottom-0 z-40 left-0 pl-8 p-4 bg-[#21262d] border border-gray-700 rounded-sm"
>
  <div
    className="flex gap-2 items-center"
  >
    <svg
      className="fill-muted w-4 h-4"
      height="16"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8.22 1.754a.25.25 0 00-.44 0L1.698 13.132a.25.25 0 00.22.368h12.164a.25.25 0 00.22-.368L8.22 1.754zm-1.763-.707c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0114.082 15H1.918a1.75 1.75 0 01-1.543-2.575L6.457 1.047zM9 11a1 1 0 11-2 0 1 1 0 012 0zm-.25-5.25a.75.75 0 00-1.5 0v2.5a.75.75 0 001.5 0v-2.5z"
        fillRule="evenodd"
      />
    </svg>
    <div>
      It will take a couple of minutes until your contract will be visible in our explorer
    </div>
  </div>
  <button
    className="pl-4 pr-0"
    onClick={[Function]}
  >
    <svg
      className="fill-muted hover:fill-blue-400 w-4 h-4"
      height="16"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3.72 3.72a.75.75 0 011.06 0L8 6.94l3.22-3.22a.75.75 0 111.06 1.06L9.06 8l3.22 3.22a.75.75 0 11-1.06 1.06L8 9.06l-3.22 3.22a.75.75 0 01-1.06-1.06L6.94 8 3.72 4.78a.75.75 0 010-1.06z"
        fillRule="evenodd"
      />
    </svg>
  </button>
</div>
`;

exports[`LoadingBar should match DOM Snapshot: and say "Please reload..."  1`] = `
<div
  className="flex fixed bottom-0 z-40 left-0 pl-8 p-4 bg-[#21262d] border border-gray-700 rounded-sm"
>
  <div
    className="flex gap-2 items-center"
  >
    <svg
      className="fill-muted w-4 h-4"
      height="16"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8.22 1.754a.25.25 0 00-.44 0L1.698 13.132a.25.25 0 00.22.368h12.164a.25.25 0 00.22-.368L8.22 1.754zm-1.763-.707c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0114.082 15H1.918a1.75 1.75 0 01-1.543-2.575L6.457 1.047zM9 11a1 1 0 11-2 0 1 1 0 012 0zm-.25-5.25a.75.75 0 00-1.5 0v2.5a.75.75 0 001.5 0v-2.5z"
        fillRule="evenodd"
      />
    </svg>
    <div>
      Please reload the page to see your new contract in our explorer
    </div>
  </div>
  <button
    className="pl-4 pr-0"
    onClick={[Function]}
  >
    <svg
      className="fill-muted hover:fill-blue-400 w-4 h-4"
      height="16"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3.72 3.72a.75.75 0 011.06 0L8 6.94l3.22-3.22a.75.75 0 111.06 1.06L9.06 8l3.22 3.22a.75.75 0 11-1.06 1.06L8 9.06l-3.22 3.22a.75.75 0 01-1.06-1.06L6.94 8 3.72 4.78a.75.75 0 010-1.06z"
        fillRule="evenodd"
      />
    </svg>
  </button>
</div>
`;
